subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -3851,11 +3851,23 @@ static int setup_cert(struct net_app_ctx *app_ctx, void *cert)
 int lwm2m_engine_start(struct lwm2m_ctx *client_ctx,
 		       char *peer_str, u16_t peer_port)
 {
+	struct sockaddr client_addr;
 	int ret = 0;
 
 	/* TODO: use security object for initial setup */
+
+	/* setup the local client port */
+	memset(&client_addr, 0, sizeof(client_addr));
+#if defined(CONFIG_NET_IPV6)
```

Comment:
This probably does not work as planned if both IPv6 and IPv4 are enabled. If peer address in peer_str is IPv4 one, then we would set the family to IPv6 here (if both IPv6 and v4 are enabled) instead of IPv4. IMHO we need to check this at runtime. Note that there is net_ipaddr_parse() function that can be used to parse IP address string and get the family.

Reply:
Yeah this part is confusing for sure.  The netapp layer doesn't use the client_addr IP type for anything other than deciding what function to call for pulling the port data (net_sin vs. net_sin6)
 in client.c line 385.   The local addr sa_family is assigned based on the remote_addr if the IP addr is unassigned (0).